src/components/views/dialogs/ReportEventDialog.tsx

```diff
@@ -0,0 +1,415 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import * as sdk from '../../../index';
+import { _t } from '../../../languageHandler';
+import { ensureDMExists } from "../../../createRoom";
+import { IDialogProps } from "./IDialogProps";
+import PropTypes from "prop-types";
+import {MatrixEvent} from "matrix-js-sdk/src/models/event";
+import {MatrixClientPeg} from "../../../MatrixClientPeg";
+import SdkConfig from '../../../SdkConfig';
+import Markdown from '../../../Markdown';
+import {replaceableComponent} from "../../../utils/replaceableComponent";
+import SettingsStore from "../../../settings/SettingsStore";
+import StyledRadioButton from "../elements/StyledRadioButton";
+
+interface IProps extends IDialogProps {
+    mxEvent: MatrixEvent;
+}
+
+interface IState {
+    // A free-form text describing the abuse.
+    reason: string,
+    busy: boolean,
+    err: string|null,
+    // If specified, the nature of the abuse, as specified by MSC3215.
+    nature: string|null,
+}
+
+
+const MODERATED_BY_STATE_EVENT_TYPE = [
+    "org.matrix.msc3215.room.moderation.moderated_by",
+    /**
+     * Unprefixed state event. Not ready for prime time.
+     *
+     * "m.room.moderation.moderated_by"
+     */
+];
+
+const ABUSE_EVENT_TYPE = "org.matrix.msc3215.abuse.report";
+
+// Standard abuse natures.
+const NATURE_DISAGREEMENT = "org.matrix.msc3215.abuse.nature.disagreement";
+const NATURE_TOXIC = "org.matrix.msc3215.abuse.nature.toxic";
+const NATURE_ILLEGAL = "org.matrix.msc3215.abuse.nature.illegal";
+const NATURE_SPAM = "org.matrix.msc3215.abuse.nature.spam";
+const NATURE_OTHER = "org.matrix.msc3215.abuse.nature.other";
+// Non-standard abuse nature.
+// It should never leave the client - we use it to fallback to
+// server-wide abuse reporting.
+const NATURE_ADMIN = "non-standard.abuse.nature.admin";
+
+/*
+ * A dialog for reporting an event.
+ *
+ * The actual content of the dialog will depend on two things:
+ *
+ * 1. Is `feature_report_to_moderators` enabled?
+ * 2. Does the room support moderation as per MSC3215, i.e. is there
+ *    a well-formed state event `m.room.moderation.moderated_by`
+ *    /`org.matrix.msc3215.room.moderation.moderated_by`?
+ */
+@replaceableComponent("views.dialogs.ReportEventDialog")
+export default class ReportEventDialog extends React.Component<IProps, IState> {
+    static propTypes = {
+        mxEvent: PropTypes.instanceOf(MatrixEvent).isRequired,
+        onFinished: PropTypes.func.isRequired,
+    };
+
```

Comment:
This can be removed as you've already declared props definition as part of `IProps`

```suggestion
```